qiskit/providers/__init__.py

```diff
@@ -110,6 +109,381 @@
    Job
    JobV1
 
+======================
+Writing a New Provider
+======================
+
+If you have a Quantum device or simulator that you would like to integrate with
+Qiskit the way to go about this is writing a provider. A provider will provide
+Terra with a method to get available :class:`~qiskit.provider.BackendV1` objects
+The :class:`~qiskit.provider.BackendV1` object provides both information
+describing a backend and its operation for the :mod:`~qiskit.transpiler` so that
+circuits can be compiled to something that is optimized and can execute on the
+backend and also the :meth:`~qiskit.providers.BackenvV1.run` method which can
+run the :class:`~qiskit.circuit.QuantumCircuit` objects and/or
+:class:`~qiskit.pulse.Schedule` objects. This enables users and other Qiskit
+APIs, such as :func:`~qiskit.execute.execute` and higher level algorithms in
+Qiskit Aqua, to get results from executing circuits on devices in a standard
+fashion regardless of how the Backend is implemented. For a simple example of
+a provider the
+`qiskit-aqt-provider <https://github.com/qiskit-community/qiskit-aqt-provider>`__
+packge is worth using as an example.
+
+Provider
+========
+
+A provider class serves a single purpose to get backend objects that enable
+executing circuits on a device or simulator. The expecation is that any
+required credentials and/or authentication will be handled in the initialization
+of a provider object. The provider object will then provide a list of backends,
+and methods to filter and acquire backends (using the provided credentials if
+required). An example provider class looks like::
+
+    from qiskit.providers import ProviderV1 as Provider
+    from qiskit.providers.providerutils import filter_backends
+
+    from .backend import MyBackend
+
+    class MyProvider(Provider):
+
+        def __init__(self, token=None):
+            super().__init__()
+            self.token = token
+            self.backends = [MyBackend(provider=self)]
+
+        def backends(self, name=None, **kwargs):
+            if name:
+                backends = [
+                    backend for backend in backends if backend.name() == name]
+            return filter_backends(backends, filters=filters, **kwargs)
+
+The key aspects to ensure are present is that any necessary information for
+authentication (if required) are present in the class and that the backends
+method matches the abstract interface and will return initialized Backend
+objects. The rest is up to the specific provider on how to implement.
```

Comment:
Big run on sentence in current form, also fairly awkward.  Needs punctuation.  Maybe use a bulleted list?